### PR TITLE
[Merged by Bors] - feat: add lemma `Quotient.exists`

### PR DESCRIPTION
**Pull request by astrainfinita**
Also `forall_quotient_iff {α : Type*} [r : Setoid α] ...` -> `Quotient.forall {α : Sort*} {s : Setoid α} ...`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Quot.lean [hunk 1]
```diff
@@ -299,15 +299,19 @@ theorem Quot.eq {α : Type*} {r : α → α → Prop} {x y : α} :
 theorem Quotient.eq [r : Setoid α] {x y : α} : Quotient.mk r x = ⟦y⟧ ↔ x ≈ y :=
   ⟨Quotient.exact, Quotient.sound⟩
 
-theorem forall_quotient_iff {α : Type*} [r : Setoid α] {p : Quotient r → Prop} :
-    (∀ a : Quotient r, p a) ↔ ∀ a : α, p ⟦a⟧ :=
-  ⟨fun h _ ↦ h _, fun h a ↦ a.induction_on h⟩
-#align forall_quotient_iff forall_quotient_iff
+lemma Quotient.forall {α : Sort*} {s : Setoid α} {p : Quotient s → Prop} :
+    (∀ a, p a) ↔ ∀ a : α, p ⟦a⟧ :=
```
Comment (eric-wieser) [inline Mathlib/Data/Quot.lean:303]: I think we should have a version for `Quotient.mk s a` too.

File: Mathlib/Data/Quot.lean [hunk 2]
```diff
@@ -299,15 +299,19 @@ theorem Quot.eq {α : Type*} {r : α → α → Prop} {x y : α} :
...
+  ⟨fun h _ ↦ h _, fun h a ↦ a.ind h⟩
+#align forall_quotient_iff Quotient.forall
+
+lemma Quotient.exists {α : Sort*} {s : Setoid α} {p : Quotient s → Prop} :
```
Comment (eric-wieser) [inline Mathlib/Data/Quot.lean:307]:
```suggestion
theorem Quotient.exists {α : Sort*} {s : Setoid α} {p : Quotient s → Prop} :
```
to match the rest of the file.

File: Mathlib/Data/Quot.lean (hunk 1 above)
Comment (negiizhao) [reply Mathlib/Data/Quot.lean:303]: Isn't `⟦a⟧` just the notation for `Quotient.mk _ a`?

Comment (eric-wieser) [reply Mathlib/Data/Quot.lean:303]: No, I think it's notation for `Quotient.mk' a`. Confusingly the meanings of `mk` and `mk'` swapped in Lean4

Comment (negiizhao) [reply Mathlib/Data/Quot.lean:303]: https://github.com/leanprover-community/mathlib4/blob/master/Mathlib/Data/Quot.lean#L223-L224

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Pull request closed without merging**